filters/auth/forwardtokenfield.go

```diff
@@ -0,0 +1,87 @@
+package auth
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"golang.org/x/net/http/httpguts"
+)
+
+const (
+	ForwardTokenFieldName = "forwardTokenField"
+)
+
+type (
+	forwardTokenFieldSpec   struct{}
+	forwardTokenFieldFilter struct {
+		HeaderName string
+		Field      string
+	}
+)
+
+// NewForwardTokenPart creates a filter to forward fields from token info or
+// token introspection or oidc user info as headers to the backend server.
+func NewForwardTokenField() filters.Spec {
+	return &forwardTokenFieldSpec{}
+}
+
+func (s *forwardTokenFieldSpec) Name() string {
+	return ForwardTokenFieldName
+}
+
+func (*forwardTokenFieldSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	headerName, ok := args[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	valid := httpguts.ValidHeaderFieldName(headerName)
```

Comment:
why not using `http.CanonicalHeaderKey(headerName)`, so we don't need the error handling here?

Reply:
According to `http.CanonicalHeaderKey(s string)` docs:
```
// If s contains a space or invalid header field bytes, it is
// returned without modifications.
```

So for me it means we still have to check result of `http.CanonicalHeaderKey(headerName)`

Reply:
Good point